yax/modules/identify_informative_hits.py

```diff
@@ -0,0 +1,217 @@
+
+    with open(identified_taxa_art, mode='r') as fh:
+        hit_taxa_lines = fh.readlines()
+
+    hit_taxa = get_hit_taxa(hit_taxa_lines)
+
+    hit_tax_ids, total_informative_hits = get_hit_taxids(hit_data,
+                                                         hit_taxa,
+                                                         lca_dist)
+
+    informative_tax_ids = get_informative_tax_ids(min_hits,
+                                                  min_relative_hits,
+                                                  hit_tax_ids,
+                                                  total_informative_hits)
+
+    hit_data.inclusion_tree = taxidtool.build_tree(hit_data.tax_id_data,
+                                                   None,
+                                                   informative_tax_ids,
+                                                   [])
+
+    hit_data.gis_to_taxids = taxidtool.build_gis_to_taxids(hit_data.
+                                                           inclusion_tree)
+
+    return hit_data
+
+
+def get_informative_tax_ids(min_hits, min_relative_hits, hit_tax_ids,
+                            total_informative_hits):
+    """Based on min_hit thresholds taxids are determined to be informative
+
+    The most stringent (highest) of min_hits and min_relative_hits is used as
+    a threshold that each tax id must meet or beat in order to determined
+    informative.
+
+    Parameters
+    ----------
+    min_hits : Int
+        absolute number of hits a taxid must exhibit to be determined
+        informative
+    min_relative_hits : Int
+        number of hits relative to total number of hits reported that a
+        taxid must exhibit to be determined informative
+    hit_tax_ids : dict
+        tax_id key to number_of_hits value representing the number of hits each
+        tax_id was observed to have in the alignmnet file
+
+    Returns
+    -------
+        list
+            tax_ids that have meet or beat minimum hit thresholds
+
+    """
+    hit_threshold = max(min_hits,
+                        (total_informative_hits/100*min_relative_hits))
+
+    informative_tax_ids = []
+    for tax_id, num_hits in hit_tax_ids.items():
+        if num_hits >= hit_threshold:
+            informative_tax_ids.append(tax_id)
+
+    return informative_tax_ids
+
+
+def get_hit_taxa(hit_taxa_lines):
+    """Builds dictionary from read_ids and the tax_ids it hit
+
+    A set of all tax_ids it hit is constructed for each read_id.
+
+    parameters
+    ----------
+        hit_taxa_lines : list
+            all lines from mg_wrapper output generated in an alignment module
+
+    Returns
+    -------
+        dict
+            read_id key to list of tax_ids value where the read was observed
+            to hit the tax_id in an alignment step
+
+    """
+    hit_taxa = {}
+
+    for line in hit_taxa_lines:
+        this_line = line.rstrip().split(":")
+        this_read_id = this_line[1]
```

Comment:
the thing i was trying to pull out... this is part of the functionality removed to be included in new understanding of mg wrapper for bowtie2